src/hidimstat/desparsified_lasso.py

```diff
@@ -152,271 +119,374 @@ def desparsified_lasso(
     if dof_ajdustement:
-        coef_max = np.max(np.abs(beta_lasso))
-        support = np.sum(np.abs(beta_lasso) > 0.01 * coef_max)
+        coef_max = np.max(np.abs(beta_reid))
+        support = np.sum(np.abs(beta_reid) > 0.01 * coef_max)
         support = min(support, n_samples - 1)
         dof_factor = n_samples / (n_samples - support)
     else:
         dof_factor = 1
 
     # Computing Desparsified Lasso estimator and confidence intervals
-    beta_bias = dof_factor * np.dot(y.T, Z) / np.sum(X * Z, axis=0)
+    # Estimating the coefficient vector
+    beta_bias = dof_factor * np.dot(y_.T, Z) / np.sum(X_ * Z, axis=0)
 
-    P = ((Z.T.dot(X)).T / np.sum(X * Z, axis=0)).T
+    # beta hat
+    P = (np.dot(X_.T, Z) / np.sum(X_ * Z, axis=0)).T
     P_nodiag = P - np.diag(np.diag(P))
     Id = np.identity(n_features)
     P_nodiag = dof_factor * P_nodiag + (dof_factor - 1) * Id
-
-    beta_hat = beta_bias - P_nodiag.dot(beta_lasso)
-
+    beta_hat = beta_bias.T - P_nodiag.dot(beta_reid.T)
+    # confidence intervals
     omega_diag = omega_diag * dof_factor**2
-    omega_invsqrt_diag = omega_diag ** (-0.5)
 
+    if not group:
+        return beta_hat, sigma_hat, omega_diag
+    else:
+        cov_hat = sigma_hat
+        if cov is not None:
+            cov_hat = cov
+        theta_hat = n_samples * inv(cov_hat)
+        return beta_hat, theta_hat, omega_diag
+
+
+def desparsified_lasso_pvalue(
+    n_samples,
+    beta_hat,
+    sigma_hat,
+    omega_diag,
+    confidence=0.95,
+    distribution="norm",
+    eps=1e-14,
+    confidence_interval_only=False,
+):
+    """
+    Calculate confidence intervals and p-values for desparsified lasso estimators.
+    This function computes confidence intervals for the desparsified lasso
+    estimator beta_hat.
+    It can also return p-values derived from these confidence intervals.
+    Parameters
+    ----------
+    n_samples : float
+        The number of samples
+    beta_hat : ndarray, shape (n_features,)
+        The desparsified lasso coefficient estimates.
+    sigma_hat : float
+        Estimated noise level.
+    omega_diag : ndarray, shape (n_features,)
+        Diagonal elements of the precision matrix estimate.
+    confidence : float, default=0.95
+        Confidence level for intervals, must be in [0, 1].
+    distribution : str, default="norm"
+        Distribution to use for p-value calculation.
+        Currently only "norm" supported.
+    eps : float, default=1e-14
+        Small value to avoid numerical issues in p-value calculation.
+    confidence_interval_only : bool, optional (default=False)
+        If True, return only confidence intervals.
+        If False, also return p-values.
+    Returns
+    -------
+    If confidence_interval_only=True:
+        cb_min : ndarray, shape (n_features,)
+            Lower bounds of confidence intervals
+        cb_max : ndarray, shape (n_features,)
+            Upper bounds of confidence intervals
```

Comment:
Comments from PR #156

Reply:
>This pattern where the output depends on the confidence interval parameter is just horrible. We should get rid of that. Having two separate functions would actually be a much better idea.

>_Originally posted by @bthirion in https://github.com/mind-inria/hidimstat/pull/156#discussion_r1990148260_

Reply:
>Moreover, this funtcion seems to have nothing to do with deparsified lasso: it just returns confidence intervals  or p-values  given estimated effect and variance, assuming that a Gaussian approximation holds. It should probably go somewhere else and be renamed.

>_Originally posted by @bthirion in https://github.com/mind-inria/hidimstat/pull/156#discussion_r1990149740_

Reply:
> > This pattern where the output depends on the confidence interval parameter is just horrible. We should get rid of that. Having two separate functions would actually be a much better idea.
> 
> > _Originally posted by @bthirion in [#156 (comment)](https://github.com/mind-inria/hidimstat/pull/156#discussion_r1990148260)_

From the issue #153, I removed this pattern to all the functions.
I don't really like to have 2 separate functions because it will duplicate the code.

Reply:
> > Moreover, this funtcion seems to have nothing to do with deparsified lasso: it just returns confidence intervals  or p-values  given estimated effect and variance, assuming that a Gaussian approximation holds. It should probably go somewhere else and be renamed.
> 
> > _Originally posted by @bthirion in [#156 (comment)](https://github.com/mind-inria/hidimstat/pull/156#discussion_r1990149740)_

It's the methods that we have associated with the desparsified lasso. 
I agree that this method can be included in the stat_tools.

Reply:
I open an issue regarding this last comment #185

Reply:
@bthirion